ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageSievingInputStream.java

```diff
@@ -13,63 +13,74 @@
-import java.io.DataInputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.nio.ByteBuffer;
+import java.io.SequenceInputStream;
 import java.util.ArrayList;
+import java.util.Enumeration;
 import java.util.List;
+import java.util.ListIterator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static com.github.ambry.messageformat.MessageFormatRecord.*;
-
 
 /**
- * InputStream that skips invalid blobs based on some validation criteria.
- * For now, the check only supports detection of message corruption
+ * InputStream that transforms messages.
  */
 public class MessageSievingInputStream extends InputStream {
-  private int validSize;
+  private int sievedStreamSize;
   private final Logger logger;
-  private ByteBuffer byteBuffer;
+  private final InputStream sievedStream;
   private boolean hasInvalidMessages;
-  private List<MessageInfo> validMessageInfoList;
+  private boolean hasDeprecatedMessages;
+  private final List<MessageInfo> sievedMessageInfoList;
+  private final List<Transformer> transformers;
 
   //metrics
-  public Histogram messageFormatValidationTime;
-  public Histogram messageFormatBatchValidationTime;
+  public final Histogram singleMessageSieveTime;
+  public final Histogram batchMessageSieveTime;
+  public final Counter messageSievingCorruptMessagesDiscardedCount;
+  public final Counter messageSievingDeprecatedMessagesDiscardedCount;
 
   /**
-   * @param stream The stream from which bytes need to be read. If the underlying stream is SocketInputStream, it needs
+   * @param inStream The stream from which bytes need to be read. If the underlying stream is SocketInputStream, it needs
    *               to be blocking
    * @param messageInfoList List of MessageInfo which contains details about the messages in the stream
-   * @param storeKeyFactory factory which is used to read the key from the stream
    * @param metricRegistry Metric register to register metrics
    * @throws java.io.IOException
    */
-  public MessageSievingInputStream(InputStream stream, List<MessageInfo> messageInfoList,
-      StoreKeyFactory storeKeyFactory, MetricRegistry metricRegistry) throws IOException {
+  public MessageSievingInputStream(InputStream inStream, List<MessageInfo> messageInfoList,
+      List<Transformer> transformers, MetricRegistry metricRegistry) throws IOException {
     this.logger = LoggerFactory.getLogger(getClass());
-    messageFormatValidationTime =
-        metricRegistry.histogram(MetricRegistry.name(MessageSievingInputStream.class, "MessageFormatValidationTime"));
-    messageFormatBatchValidationTime = metricRegistry.histogram(
-        MetricRegistry.name(MessageSievingInputStream.class, "MessageFormatBatchValidationTime"));
-    validSize = 0;
+    this.transformers = transformers;
+    singleMessageSieveTime = metricRegistry.histogram(
+        MetricRegistry.name(com.github.ambry.messageformat.MessageSievingInputStream.class, "SingleMessageSieveTime"));
```

Comment:
minor: it seems like the `com.github.ambry.messageformat.` is not required (it's this class?)

Reply:
will fix.